### PR TITLE
Update chooser widget styles across all types

### PR DESCRIPTION
**Pull request by thibaudcolas**
Implements the [new chooser widget styles](https://www.figma.com/file/yHSyon6x5aeUn5RDTDYfF6/Wagtail-Design-System-(Copy)?node-id=5172%3A30787) as part of the page editor redesign.

## Scenarios

There are essentially two styles. One for:

- Generic / base (`BaseChooser`, and legacy `AdminChooser`)
- Page
- Snippet
- Document
- Task

Those all look almost the same. The only difference between them is the icon, used for the chosen state, and for the add button’s icon.

Chosen state:

document-chooser

Add button:

chooser-add

Then there is another slightly different style for the chosen state of images:

image-chooser

This isn’t very close to the designs. I foresee implementation issues with the current designs so decided to keep this simple until I check with Ben.

There is a third style when choosers are used in an InlinePanel with each element being a chooser and nothing else, which I’ve left out of this PR.

---

Finally… the promoted search terms’ UI was also using the choosers’ markup for some reason. I’ve done minimal changes to it so it retains its current layout.

## Implementation

Back to the real choosers, I kept the markup very similar to what it was before, with the notable exception of the title – which now uses a BEM class, a data- attribute, and an id so it can be referenced in `aria-describedby`.

Another notable change is the removal of the custom widget for the workflow task chooser. As far as I can tell it doesn’t need a custom template anymore.

Compared to the designs, another notable difference is with the buttons – they’ll be of a slightly different size (height?) as we haven’t updated them yet.

---

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
-   ~~[ ] For Python changes: Have you added tests to cover the new/fixed behaviour?~~
-   [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [x] **Please list the exact browser and operating system versions you tested**: Safari 15.5 macOS 12.4, Chrome 103 macOS 12.4, Firefox 102 macOS 12.4
    -   [x] **Please list which assistive technologies [^3] you tested**: VoiceOver, Windows high contrast mode
-   [x] For new features: Has the documentation been updated accordingly?

### CODE REVIEW

File: client/scss/components/_chooser.scss [hunk 1]
```diff
@@ -1,131 +1,92 @@
...
-TODO this chooser style has been made more generic based on two identical
-methods for choosing pages and images that were previously included in their
-own less files in each app directory (and since deleted). It would be best if
-an admin 'theme' provided all the design for a UI in a single place, but
-should that be a series of overrides to the css provided from an app? If so,
-perhaps those two previous less files should be re-instated and then
-overridden here? hmm.
-*/
-
 .chooser {
```
Comment (lb-) [inline client/scss/components/_chooser.scss:1]: `w-chooser` ? 😀

Comment (thibaudcolas) [reply client/scss/components/_chooser.scss:1]: I’ve tried to leave refactorings to a minimum for this one, we’re way past schedule 🙈

File: client/scss/layouts/_groups-edit.scss [hunk 2]
```diff
@@ -24,9 +24,3 @@
     padding: 0;
   }
 }
-
-.page-permissions-listing {
-  .admin_page_chooser .field-content {
-    width: 100%; // so that 'choose another page' button displays in its entirety
-  }
-}
```
Comment (thibaudcolas) [inline client/scss/layouts/_groups-edit.scss:32]: As far as I can tell this is no longer needed with the short button labels (same for workflows).

File: wagtail/admin/templates/wagtailadmin/widgets/chooser.html [hunk 3]
```diff
@@ -7,32 +7,36 @@
...
+            <div class="chooser__preview" role="presentation">
+                {% if icon %}{% icon name=icon %}{% endif %}
+            </div>
+        {% endblock chosen_icon %}
         {% block chosen_state_view %}
-            <span class="title">{{ display_title }}</span>
+            <div class="chooser__title" data-chooser-title id="{{ title_id }}">{{ display_title }}</div>
         {% endblock %}
 
-        <ul class="actions">
```
Comment (thibaudcolas) [inline wagtail/admin/templates/wagtailadmin/widgets/chooser.html:18]: I haven’t really changed the actions, they’re just in a different order, and have an additional `aria-describedby` attribute.

File: wagtail/admin/widgets/workflows.py [hunk 4]
```diff
@@ -10,12 +10,9 @@
 
 class AdminTaskChooser(BaseChooser):
     choose_one_text = _("Choose a task")
-    choose_another_text = _("Choose another task")
-    link_to_chosen_text = _("Edit this task")
     model = Task
-    template_name = "wagtailadmin/workflows/widgets/task_chooser.html"
+    icon = "thumbtack"
     chooser_modal_url_name = "wagtailadmin_workflows:task_chooser"
-    classname = "task-chooser"
```
[the remark this answers is not in the document]
Comment (lb-) [reply wagtail/admin/widgets/workflows.py:18]: We should be careful with this - having these built-in classes really helps those that are customising the appearance.

If it's ok I will add these back in before merging.

**Comment by lb-** [comment]
Hope I did not jump the gun here @gasman - I just remembered that you were also planning to review.

**Pull request merged**